Tests/FluentAssertions.Specs/Events/EventAssertionSpecs.cs

```diff
@@ -971,6 +972,151 @@ public void One_matching_argument_type_with_two_or_more_parameters_matching_a_mi
         }
     }
 
+    public class MonitorDefaultBehavior
+    {
+        [Fact]
+        public void Broken_event_add_accessors_fails()
+        {
+            // Arrange
+            var sut = new TestEventBrokenEventHandlerRaising();
+
+            // Act / Assert
+            sut.Invoking(c =>
+            {
+                using var monitor = c.Monitor<IAddFailingEvent>();
+            }).Should().Throw<TargetInvocationException>();
+        }
+
+        [Fact]
+        public void Broken_event_remove_accessors_fails()
+        {
+            // Arrange
+            var sut = new TestEventBrokenEventHandlerRaising();
+
+            // Act / Assert
+            sut.Invoking(c =>
+            {
+                using var monitor = c.Monitor<IRemoveFailingEvent>();
+            }).Should().Throw<TargetInvocationException>();
+        }
+    }
+
+    public class IgnoreMisbehavingEventAccessors
+    {
+        [Fact]
+        public void Monitoring_class_with_broken_event_add_accessor_succeeds()
+        {
+            // Arrange
+            var classToMonitor = new TestEventBrokenEventHandlerRaising();
+
+            // Act / Assert
+            classToMonitor.Invoking(c =>
+            {
+                using var monitor = c.Monitor<IAddFailingEvent>(opt => opt.IgnoreEventAccessorExceptions());
+            }).Should().NotThrow();
+        }
+
+        [Fact]
+        public void Class_with_broken_event_remove_accessor_succeeds()
+        {
+            // Arrange
+            var classToMonitor = new TestEventBrokenEventHandlerRaising();
+
+            // Act / Assert
+            classToMonitor.Invoking(c =>
+            {
+                using var monitor = c.Monitor<IRemoveFailingEvent>(opt => opt.IgnoreEventAccessorExceptions());
+            }).Should().NotThrow();
+        }
+
+        [Fact]
+        public void Recording_event_with_broken_add_accessor_succeeds()
+        {
+            // Arrange
+            var classToMonitor = new TestEventBrokenEventHandlerRaising();
+
+            using var monitor = classToMonitor.Monitor<IAddFailingEvent>(opt => opt.IgnoreEventAccessorExceptions(true));
+
+            //Act
+            classToMonitor.RaiseOkEvent();
+
+            //Assert
+            monitor.MonitoredEvents.Should().HaveCount(1);
+        }
+
+        [Fact]
+        public void Ignoring_broken_event_accessor_should_also_not_record_events()
+        {
+            // Arrange
+            var classToMonitor = new TestEventBrokenEventHandlerRaising();
+
+            using var monitor = classToMonitor.Monitor<IAddFailingEvent>(opt => opt.IgnoreEventAccessorExceptions(false));
```

Comment:
`IgnoreEventAccessorExceptions` now takes a Boolean parameter which we'd like to avoid in the APIs.
See https://csharpcodingguidelines.com/maintainability-guidelines/#AV1564 for an explanation.

I assume you combined them, as `recordEventsWithBrokenAccessor` only has an effect when ignoring event accessor exceptions?

Reply:
@apazureck I think this one is still open :)

Reply:
Added another method to set recroding to true, even if event setter fails.